src/core/resources/gas/estimateApprovalGasLimit.ts

```diff
@@ -0,0 +1,139 @@
+import { useQuery } from '@tanstack/react-query';
+import { Address } from 'viem';
+
+import { estimateApprove } from '~/core/raps/actions';
+import { estimateERC721Approval } from '~/core/raps/actions/unlock';
+import {
+  QueryConfig,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+  createQueryKey,
+  queryClient,
+} from '~/core/react-query';
+import { gasUnits } from '~/core/references/gasUnits';
+import { ChainId } from '~/core/types/chains';
+
+// ///////////////////////////////////////////////
+// Query Types
+
+export type EstimateAprovalGasLimitArgs = {
+  chainId: ChainId;
+  ownerAddress: Address;
+  assetAddress?: Address;
+  spenderAddress?: Address;
+  assetType: 'erc20' | 'erc721';
```

Comment:
no support for erc1155?

Reply:
this is just a local param 

https://github.com/rainbow-me/browser-extension/pull/1322/files#diff-ecd89036959c55e3706a5919f309eba47e8e25aaef429bdda06fcb2cc83eefebR101

backend is returning 'nft' so i'm gonna do the same to have it cleaner